src/pages/posts/[postid].astro

```diff
@@ -33,27 +37,46 @@ if (data.length > 1) {
 
 const content = data[0];
 
-const trimmedText = content.body.slice(0, 100) + (content.body.length > 100 ? '...' : '');
+const trimmedText =
+    content.body.slice(0, 100) + (content.body.length > 100 ? '...' : '');
 
 const html = sanitizeHtml(await marked.parse(content.body), {
     allowedTags: sanitizeHtml.defaults.allowedTags.concat(['img']),
 });
 
-const showCopyLinkWarning = Astro.url.searchParams.get('created') !== null;
-const url = Astro.url.href;
+const showCopyLinkWarning = Astro.url.searchParams.get('created') === content.edit_key;
+const url = Astro.url.href.split('?')[0];
+const editURL = url + `/edit?key=${content.edit_key}`;
 ---
 
 <Seo title={content.title} description={trimmedText} />
 <BaseLayout title={`${content.title || 'Untitled'}`}>
-
-    {showCopyLinkWarning && 
-    <section class="attention">
-        <div>
-            <div>Post published successfully!</div>
-            {content.unlisted && <div>This post is <em>unlisted</em> - make sure to copy the link so you can get back here.</div>}
-        </div>
-        <button id="copyURL">Copy URL</button>
-    </section>
+    {
+        showCopyLinkWarning && (
+            <section class="attention">
+                <div>
+                    <div>Post published successfully!</div>
+                    {content.unlisted && (
+                        <div>
+                            This post is <em>unlisted</em> - make sure to copy
+                            the link so you can get back here.
+                        </div>
+                    )}
+                </div>
+                <div class="right">
+                    <CopyToClipboard
+                        client:only="preact"
+                        type="success"
+                        data={url}
+                    >
+                        Copy URL
+                    </CopyToClipboard>
+                    <CopyToClipboard client:only="preact" type="info" data={editURL}>
+                        Copy edit URL
```

Comment:
I wonder if this would make more sense as an 'Edit' button. Then we just tell users to copy the url for the 'created' page to access that button, instead of needing to copy the link and then revisit it to edit.

So you'd have [Copy Public URL] and [Copy Private URL], with a little explainer for both.

Reply:
I think it could lead to a little bit of confusion, but also think it's a better track than what we have atm. Let's keep this for now and discuss more, maybe kick some ideas off friends?